regolith/dates.py

```diff
@@ -166,49 +170,82 @@ def get_dates(thing):
 
     If begin_year is found, the begin month and begin day are missing they are set to
     1 and 1, respectively
+
+    If a date field prefix is passed in this function will search for prefix_year as
+    well as prefix_month, prefix_day, and prefix_date. For example, if the prefix string
+    passed in is "submitted" then this function will look for submitted_date instead of
+    just date.
+
+    Examples
+    --------
+    >>> get_dates({'submission_day': 10, 'submission_year': 2020, 'submission_month': 'Feb'}, "submission")
+
+    This would return a dictionary consisting of the begin date, end, date, and date for the given input.
+    Instead of searching for "day" in the thing, it would search for "submission_day" since a prefix was
+    given. The following dictionary is returned (note that a submission_date and a date key are in the
+    dictionary):
+    {'begin_date': datetime.date(2020, 2, 10),
```

Comment:
is this correct or does it return ``None`` for begin_date in this case?  this seems be ambiguous (see above in the docstring where it shows that if it finds a well defined date it returns date but not begin/end_date)

Reply:
This is correct I took the example right from one of the tests